### DISCUSSION TITLE
🗂 User access in business app data

### DISCUSSION THREAD

**Discussion by mohsenxad** [lettucesocial/com-lettucesocial-customer-panel#57] [Ideas]
### 🗂 User access in business app data

We have 3 types of resources in business app:

1. Search with zip code feature 
2. See search result list 
3. See search result list item
---
Access Level for **search with zip code**
- no limit on search count (how many times each user can search for a zip code)
- 3 search result limitation for unknown user (each user can search 3 times and then should request for more access)
- search for nearby business zip code
---
Access Level for **search result list**
- no search result limitation 
- show 3 search result visibility 
---
Access Level for **search result list item**

- active visit creator Instagram link
- de active visit creator Instagram link

**Comment by maxzad** [comment]
@mohsenxad - here are my initial thoughts. We can chat more during our meeting on Tuesday.

User access level - MVP

https://www.loom.com/share/ad97c187b47c4b309927044ed4436553?sid=468ef129-d7b0-4233-b26b-e72a0109b3a3

------------------
User access level - Ideal state

https://www.loom.com/share/83a8dff48a724ab4b8a36eebd159833c?sid=a6ad42a7-b411-4ffc-b5e6-a6cd106d1a13

-------
Link to the Google Slide 

https://docs.google.com/presentation/d/1WRbxDLNMi7ZlNz36lgKd8eCqWbw_hHUsFcu6MaASJ-U/edit?usp=sharing

**Comment by mohsenxad** [comment]
@maxzad 

📃 **limited search list result**

**solution 1**
for 4th item of search list item
if unknown user click on "collaborate" button, 🤔**why we should navigate the user to login page?**
🎯 we push the user to finish the order and payment with the selected creator even by just seeing the profile image

and if click on view Instagram page , we show the login prompt

**solution 2**
all the creators after first 3 in list, show in a box 
this box have a message linke " there are 36 more creator that need to be login for visit more data"
with one login button that will navigate user to login page

not for each creator, have on link and blurred name and city

**Reply by maxzad** [reply]
@mohsenxad 

I like Solution 1
I am fine with getting the user to go through the purchase process, but after the deposit authorization is successful, then, we should somehow share the IG handle of the creator they just asked to collab with. It could be included in the confirmation email.

**Reply by maxzad** [reply]
@mohsenxad I changed my mind on this. It really doesn’t make sense to show the client the creator's IG handle after authorizing the payment. What if they find the creator disappointing or find her feed not aligned with their business or values? This is not just hypothetical. It happened to me. I had a conversation with two business owners recently where we reviewed the list of creators in their zip code, and the business owner would say, I looked at Jane’s feed, and it is not what we are looking for. I think we should not do this. It will create a bad customer experience.

**Comment by mohsenxad** [comment]
@maxzad
 
in both steps/solutions (MVP and Ideal)  we need signup flow, so i prefer to implement the Ideal solution. implementing Ideal solution takes same time as MVP one.

**Comment by mohsenxad** [comment]
@maxzad 

when a signed up business, reached 5 search zip code limit,

we can't show "Login" message

what should he/she see in UI of search page?

**Reply by maxzad** [reply]
@mohsenxad 
see my comments below.

**Comment by mohsenxad** [comment]
@maxzad 

when we show to any type of user this view👇

![image](https://github.com/lettucesocial/com-lettucesocial-customer-panel/assets/3274057/83e9775f-99ab-4e73-9d22-537dc2ac8e88)

we prevent he/she to search a previous zip code that had been searched before

🥪 example :
 I'm a local business that have searched for 5 zip codes, now a want to see the result of my business zip code again to order a collaboration. but when i come to start searching my business zip code, view restricted page

I think we have to change the UX of search limitation restriction

**Reply by maxzad** [reply]
@mohsenxad 
Let's back up and clarify access levels and search result limitations for each.
Access levels: Level 1 (unknown user) - Level 2 (signed up but not approved) - Level 3 (signed up and approved)

Search zip code limitations:
Level 1: They can only search 3 unique zip codes, which means if they search the same zip code 5 times, it still counts as 1 unique zip code. These 3 unique zip codes could be all over the country since we don't have their own business zip code, so we can restrict them to 3 zip codes to their location, and we don't have access to their GPS to use that as a reference point.

Levels 2 and 3: They can only search 5 unique zip codes. One is their own business zip code, and the other 4 are the four closest zip codes to their own zip code. This is less of a zip code search restriction and more of restricting them from searching all over the country.

In the case of level 1:
We show them the UX you have up there. They need to Sign up or login.

In the case of levels 2 and 3, the message should read.
"The lettuceSocial app restricts users to searching for creators within their community to build a hyper-local, community-powered marketing channel for small businesses. You can search the following zip codes:"

then, underneath the message, we can show the list of the 5 zip codes the user can search for in the form of a link. The top one should be their own business zip code. The example below for a business located in 91320 who is trying to search any other zip code outside of the following 5 zip codes.

![V1- LettuceSocial Business Mobile App Design-Signed In User Searching Outside 5 Zip codes](https://github.com/lettucesocial/com-lettucesocial-customer-panel/assets/19561243/206f2b89-cf3d-4887-b9f6-51289895707f)
